### PR TITLE
chore: cleaned up guava and jakarta.websocket dependencies to handle the version in one place.

### PR DESCRIPTION
**Pull request by czp13**
## Description
Cleaned up the above dependencies version handling (guava and jakarta web socket).

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,6 +95,8 @@
         <jakarta.validation.version>3.0.2</jakarta.validation.version>
         <jakarta.annotation.api.version>2.1.1</jakarta.annotation.api.version>
         <jaxb.version>4.0.3</jaxb.version>
+        <guava.version>32.1.0-jre</guava.version>
+        <jakarta.websocket>2.1.1</jakarta.websocket>
```
Comment (mcollovati) [inline pom.xml:99]:
```suggestion
        <jakarta.websocket.version>2.1.1</jakarta.websocket.version>
```

File: vaadin-dev-server/pom.xml [hunk 2]
```diff
@@ -65,13 +65,13 @@
         <dependency>
             <groupId>jakarta.websocket</groupId>
             <artifactId>jakarta.websocket-api</artifactId>
-            <version>2.1.1</version>
+            <version>${jakarta.websocket}</version>
```
Comment (mcollovati) [inline vaadin-dev-server/pom.xml:68]:
```suggestion
            <version>${jakarta.websocket.version}</version>
```

File: vaadin-dev-server/pom.xml [hunk 3]
```diff
@@ -65,13 +65,13 @@
...
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>jakarta.websocket</groupId>
             <artifactId>jakarta.websocket-client-api</artifactId>
-            <version>2.1.1</version>
+            <version>${jakarta.websocket}</version>
```
Comment (mcollovati) [inline vaadin-dev-server/pom.xml:74]:
```suggestion
            <version>${jakarta.websocket.version}</version>
```

File: flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml [hunk 4]
```diff
@@ -72,13 +72,13 @@
         <dependency>
             <groupId>jakarta.websocket</groupId>
             <artifactId>jakarta.websocket-api</artifactId>
-            <version>2.1.1</version>
+            <version>${jakarta.websocket}</version>
```
Comment (mcollovati) [inline flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml:75]:
```suggestion
            <version>${jakarta.websocket.version}</version>
```

File: flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml [hunk 5]
```diff
@@ -72,13 +72,13 @@
...
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>jakarta.websocket</groupId>
             <artifactId>jakarta.websocket-client-api</artifactId>
-            <version>2.1.1</version>
+            <version>${jakarta.websocket}</version>
```
Comment (mcollovati) [inline flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml:81]:
```suggestion
            <version>${jakarta.websocket.version}</version>
```

**Review by mcollovati — CHANGES_REQUESTED** [review]

File: flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml (hunk 5 above)
Comment (czp13) [reply flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml]: Thank you @mcollovati , fixed it with a separate commit (was easier to rename the variable, than commit the suggestion 5 times, that is why a separate commit).

File: flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml (hunk 4 above)
Comment (czp13) [reply flow-tests/test-frontend/vite-embedded-webcomponent-resync/pom.xml]: Thank you! Should be fixed!

File: vaadin-dev-server/pom.xml (hunk 3 above)
Comment (czp13) [reply vaadin-dev-server/pom.xml]: Thank you! Should be fixed!

File: vaadin-dev-server/pom.xml (hunk 2 above)
Comment (czp13) [reply vaadin-dev-server/pom.xml]: Thank you! Should be fixed!

File: pom.xml (hunk 1 above)
Comment (czp13) [reply pom.xml]: Thank you! Should be fixed!

**Review by mcollovati — APPROVED** [review]

**Pull request merged**